daemon/src/xtra_ext.rs

```diff
@@ -50,3 +52,42 @@ where
         Ok(())
     }
 }
+
+#[async_trait]
+pub trait SendInterval<A, M>
+where
+    M: Message,
+    A: xtra::Handler<M>,
+{
+    /// Similar to xtra::Context::notify_interval, however it uses `send`
+    /// instead of `do_send` under the hood.
+    /// The crucial difference is that this function waits until previous
+    /// handler returns before scheduling a new one, thus preventing them from
+    /// piling up.
+    /// As a bonus, this function is non-fallible.
+    async fn send_interval<F>(self, duration: Duration, constructor: F)
+    where
+        F: Send + Sync + Fn() -> M,
+        M: Message,
```

Comment:
Perhaps restrict this to handlers that return `()`?


```suggestion
        M: Message<Result = ()>,
```

Reply:
great suggestion, thanks!